internal/pkg/vm/internal/message/result.go

```diff
@@ -41,9 +30,13 @@ func Ok() Receipt {
 //
 // Callers do NOT need to encode the value before calling this method.
 func Value(obj interface{}) Receipt {
-	aux, err := encoding.Encode(obj)
-	if err != nil {
-		return Receipt{ExitCode: exitcode.SysErrSerialization}
+	var aux []byte
+	if obj != nil {
+		var err error
+		aux, err = encoding.Encode(obj)
+		if err != nil {
+			return Receipt{ExitCode: exitcode.SysErrSerialization}
```

Comment:
is the caller responsible for adding the other fields -- gas and return value?

Reply:
Cleaned up